[dynamo, guards] cache GuardBuilder.get() on sources

williamwen42 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #168886
* #168386
* __->__ #168203
* #168131

Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 16s -> 9s on a local tlparse. On the guard build benchmark, time went from 84.66s -> 57.25s.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @Lucaskabela @jataylo @chenyang78

williamwen42 committed:
[dynamo, guards] cache GuardBuilder.get() on sources

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, guards] cache GuardBuilder.get() on sources"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, guards] cache GuardBuilder.get() on sources"


Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 16s -> 9s locally.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, guards] cache GuardBuilder.get() on sources"


Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 16s -> 9s locally.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, guards] cache GuardBuilder.get() on sources"


Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 16s -> 9s locally.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, guards] cache GuardBuilder.get() on sources"


Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 16s -> 9s locally.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, guards] cache GuardBuilder.get() on sources"


Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 16s -> 9s locally.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, guards] cache GuardBuilder.get() on sources"


Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 16s -> 9s on a local tlparse. On the guard build benchmark, time went from 84.66s -> 57.25s.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

williamwen42 committed:
Update

[ghstack-poisoned]

anijain2305 reviewed (commented):
The main missing point for me - Its unclear why we need `_name_template` - can we add comments ?

williamwen42 committed:
Update

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, guards] cache GuardBuilder.get() on sources"


Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 16s -> 9s on a local tlparse. On the guard build benchmark, time went from 84.66s -> 57.25s.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, guards] cache GuardBuilder.get() on sources"


Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 16s -> 9s on a local tlparse. On the guard build benchmark, time went from 84.66s -> 57.25s.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, guards] cache GuardBuilder.get() on sources"


Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 16s -> 9s on a local tlparse. On the guard build benchmark, time went from 84.66s -> 57.25s.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, guards] cache GuardBuilder.get() on sources"


Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 16s -> 9s on a local tlparse. On the guard build benchmark, time went from 84.66s -> 57.25s.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]
